fhir-registry/src/main/java/com/ibm/fhir/registry/FHIRRegistry.java

```diff
@@ -177,13 +178,10 @@ public String getDefaultVersion(String url, Class<? extends Resource> resourceTy
     public <T extends Resource> Collection<T> getResources(Class<T> resourceType) {
         Objects.requireNonNull(resourceType);
         requireDefinitionalResourceType(resourceType);
-        List<T> resources = new ArrayList<>();
-        for (FHIRRegistryResourceProvider provider : providers) {
-            for (FHIRRegistryResource registryResource : provider.getRegistryResources(resourceType)) {
-                resources.add(resourceType.cast(registryResource.getResource()));
-            }
-        }
-        return Collections.unmodifiableList(resources);
+        return providers.parallelStream()
+                .flatMap(p -> p.getRegistryResources(resourceType).stream())
+                .map(rr -> resourceType.cast(rr.getResource()))
+                .collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
```

Comment:
@JohnTimm didn't you find this was less efficient? is this OK?  (semantically correct)

Reply:
i think John found that streams were a tad slower than loops in general.  however, in this particular case, the parallelStream should let the registry as all the RegistryResourceProviders for all their resources in parallel.  I think it should be faster when multiple providers are in play (especially if each one needs to read its own index file like in the case of a server start).

Reply:
I'm not opposed to this change but would like to use benchmarking to determine if it helps. When I switched out the streams to standard loops it was driven by some benchmarking results and some hot spots I found with VisualVM

Reply:
ok, I backed it out since it doesn't really belong with this PR anyway